quinn-proto/src/connection/assembler.rs

```diff
@@ -60,13 +45,16 @@ impl Assembler {
     }
 
     // Get the the next ordered chunk
-    pub(crate) fn read_chunk(&mut self) -> Result<Option<Bytes>, IllegalOrderedRead> {
+    pub(crate) fn read(
+        &mut self,
+        max_length: Option<usize>,
```

Comment:
Is there a benefit to this being `Option` rather than allowing indifferent callers to pass in `usize::MAX`?

Reply:
Benefits I could think of: maybe more obvious that it's a sentinel value, `None` is shorter to write than `usize::MAX`. But these arguments weren't enough to convince me that the simpler type wasn't more attractive, so changed it as you proposed.

Reply:
I'm not even sure I'd call it a sentinel value; the beauty of it is that no special logic is needed, so we actually have *fewer* branches.